pkg/vulnsrc/vulnerability/vulnerability.go

```diff
@@ -20,16 +20,41 @@ var (
 	getVulnerabilityDetailFunc = db.Config{}.GetVulnerabilityDetail
 )
 
-func GetDetail(vulnID string) (types.Severity, types.VendorSeverity, string, string, []string) {
+func GetDetail(vulnID string) (types.Severity, types.VendorSeverity, types.VendorVectors, string, string, []string) {
 	details, err := getVulnerabilityDetailFunc(vulnID)
 	if err != nil {
 		log.Println(err)
-		return types.SeverityUnknown, nil, "", "", nil
+		return types.SeverityUnknown, nil, nil, "", "", nil
 	} else if len(details) == 0 {
-		return types.SeverityUnknown, nil, "", "", nil
+		return types.SeverityUnknown, nil, nil, "", "", nil
 	}
 
-	return getSeverity(details), getVendorSeverity(details), getTitle(details), getDescription(details), getReferences(details)
+	return getSeverity(details), getVendorSeverity(details), getVendorVectors(details), getTitle(details), getDescription(details), getReferences(details)
+}
+
+func getVendorVectors(details map[string]types.VulnerabilityDetail) types.VendorVectors {
+	vv := make(types.VendorVectors)
+	for vendor, detail := range details {
+		VectorV2 := detail.CvssVector
+		VectorV3 := detail.CvssVectorV3
+
+		switch {
+		case VectorV2 != "" && VectorV3 != "":
+			vv[vendor] = map[string]string{
+				"v2": VectorV2,
+				"v3": VectorV3,
+			}
+		case VectorV2 != "" && VectorV3 == "":
+			vv[vendor] = map[string]string{
+				"v2": VectorV2,
+			}
+		case VectorV2 == "" && VectorV3 != "":
+			vv[vendor] = map[string]string{
+				"v3": VectorV3,
+			}
+		}
+	}
```

Comment:
If you add `json:omitempty`, these lines are not necessary, right?

```suggestion
		vv[vendor = CVSSVector {
			V2: VectorV2,
			V3: VectorV3,
		}
```

Reply:
It is still needed since it's a nested struct, because omitempty won't work on it. You can see this test will fail if you remove this logic: https://github.com/aquasecurity/trivy-db/pull/43/files#diff-bc5f6ed1d12bdab4d1aeca18ca3003d3R24

I also agree that I don't like this logic and would like to get rid of it, do you have another idea?

Reply:
I think it is simple to fix it. You just need to ignore the case when both values are empty.

```
		if detail.CvssVector == "" && detail.CvssVectorV3 == "" {
			continue
		}
		vv[vendor] = types.CVSSVector{
			V2: detail.CvssVector,
			V3: detail.CvssVectorV3,
		}
```

Reply:
Thanks, that did the trick. I pushed an update: https://github.com/aquasecurity/trivy-db/pull/43/commits/a48e663c7d4669fa065b049fd2cab611e1693e39